File: WorkIt/app/src/main/java/prot3ct/workit/views/assigned_tasks/AssignedTasksActivity.java
package prot3ct.workit.views.assigned_tasks;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.SearchView;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;

import prot3ct.workit.R;
import prot3ct.workit.views.assigned_tasks.base.AssignedTasksContract;
import prot3ct.workit.views.navigation.DrawerUtil;

public class AssignedTasksActivity extends AppCompatActivity {
    private AssignedTasksContract.Presenter presenter;
    private AssignedTasksFragment assignedTasksFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_assigned_tasks);
        Toolbar toolbar = this.findViewById(R.id.id_drawer_toolbar);
        setSupportActionBar(toolbar);
        setTitle("Assigned Tasks");
        DrawerUtil drawer = new DrawerUtil(this, toolbar);
        drawer.getDrawer();

        assignedTasksFragment =
                (AssignedTasksFragment) getSupportFragmentManager().findFragmentById(R.id.fragment_container);

        if (assignedTasksFragment == null) {
            assignedTasksFragment = AssignedTasksFragment.newInstance();

            getSupportFragmentManager()
                    .beginTransaction()
                    .add(R.id.fragment_container, assignedTasksFragment)
                    .commit();
        }

        this.presenter = new AssignedTasksPresenter(assignedTasksFragment, this);
        assignedTasksFragment.setPresenter(this.presenter);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_options, menu);

        MenuItem ourSearchItem = menu.findItem(R.id.menu_search);

        SearchView sv = (SearchView) ourSearchItem.getActionView();
        sv.setQueryHint("Search by title");
        sv.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                return true;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                assignedTasksFragment.filterTask(newText);
                return true;
            }
        });
        return true;
    }
}

File: WorkIt/app/src/main/java/prot3ct/workit/views/assigned_tasks/AssignedTasksActivity.kt
package prot3ct.workit.views.assigned_tasks

import androidx.appcompat.app.AppCompatActivity
import prot3ct.workit.views.assigned_tasks.base.AssignedTasksContract
import prot3ct.workit.views.assigned_tasks.AssignedTasksFragment
import android.os.Bundle
import android.view.Menu
import prot3ct.workit.R
import prot3ct.workit.views.navigation.DrawerUtil
import prot3ct.workit.views.assigned_tasks.AssignedTasksPresenter
import android.view.MenuInflater
import androidx.appcompat.widget.SearchView
import androidx.appcompat.widget.Toolbar
import prot3ct.workit.views.chat.ChatFragment

class AssignedTasksActivity : AppCompatActivity() {

    private lateinit var presenter: AssignedTasksContract.Presenter
    private var assignedTasksFragment: AssignedTasksFragment? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_assigned_tasks)
        val toolbar = findViewById<Toolbar>(R.id.id_drawer_toolbar)
        setSupportActionBar(toolbar)
        title = "Assigned Tasks"
        val drawer = DrawerUtil(this, toolbar)
        drawer.getDrawer()
        assignedTasksFragment =
            supportFragmentManager.findFragmentById(R.id.fragment_container) as AssignedTasksFragment?

        if (assignedTasksFragment == null) {
            assignedTasksFragment = AssignedTasksFragment.newInstance()
            supportFragmentManager
                .beginTransaction()
                .add(R.id.fragment_container, assignedTasksFragment!!)
                .commit()
        }

        presenter = AssignedTasksPresenter(assignedTasksFragment!!, this)
        assignedTasksFragment!!.setPresenter(presenter)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.menu_options, menu)
        val ourSearchItem = menu.findItem(R.id.menu_search)
        val sv = ourSearchItem.actionView as SearchView
        sv.queryHint = "Search by title"
        sv.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                return true
            }

            override fun onQueryTextChange(newText: String): Boolean {
                assignedTasksFragment!!.filterTask(newText)
                return true
            }
        })
        return true
    }
}